src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -26,21 +26,70 @@ public PeWritingException(Exception inner)
 
     internal static class PeWriter
     {
-        public static bool WritePeToStream(
+        internal static bool WritePeToStream(
+             EmitContext context,
+             CommonMessageProvider messageProvider,
+             Func<Stream> getPeStream,
+             Func<Stream> getMetadataPeStreamOpt,
+             Func<Stream> getPortablePdbStreamOpt,
+             PdbWriter nativePdbWriterOpt,
+             string pdbPathOpt,
+             bool metadataOnly,
+             bool isDeterministic,
+             CancellationToken cancellationToken)
+        {
+            if (!WritePeToStream(
+                context,
+                messageProvider,
+                getPeStream,
+                getPortablePdbStreamOpt,
+                nativePdbWriterOpt,
+                pdbPathOpt,
+                metadataOnly,
+                isDeterministic,
+                cancellationToken))
+            {
+                return false;
+            }
+
+            // produce the secondary output (ref assembly) if needed
+            if (getMetadataPeStreamOpt != null)
+            {
+                Debug.Assert(!metadataOnly);
+
+                if (!WritePeToStream(
+                    context,
+                    messageProvider,
+                    getMetadataPeStreamOpt,
+                    getPortablePdbStreamOpt: null,
+                    nativePdbWriterOpt: null,
+                    pdbPathOpt: null,
+                    metadataOnly: true,
+                    isDeterministic: isDeterministic,
+                    cancellationToken: cancellationToken))
+                {
+                    return false;
+                }
+            }
```

Comment:
I'm not sure I see the advantage. There is a single caller, but why have two methods with same name spread across two classes? Especially since this one is a thin wrapper on top of the existing one.